### PR TITLE
Remove ML(torch/tf) libraries and its dependencies

### PR DESCRIPTION
**Pull request by 86sanj**
# Peer Review Information
This PR includes following changes:

1. Remove tf and torch dependencies
2. Remove any public dataset
3. Remove any redundant library
4. Remove loader, draft notebook, protos
4. Updated readme
5. Updated docs

Requirement: https://github.com/Unity-Technologies/datasetinsights/issues/143
# Pull Request Check List

- [x] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.

### CODE REVIEW

File: datasetinsights/datasets/coco.py [hunk 1]
```diff
@@ -1,22 +1,12 @@
 import fcntl
```
Comment (adason) [inline datasetinsights/datasets/coco.py]: Do we need to keep coco.py file here? Are we keeping the dataset class in this repo?

File: datasetinsights/datasets/groceries_real.py [hunk 2]
```diff
@@ -4,20 +4,15 @@
...
 
 import datasetinsights.constants as const
 from datasetinsights.io.bbox import BBox2D
-from datasetinsights.io.gcs import GCSClient
 
 from .base import Dataset
 from .exceptions import DatasetNotFoundError
-from .protos import string_int_label_map_pb2
 
 logger = logging.getLogger(__name__)
```
Comment (adason) [inline datasetinsights/datasets/groceries_real.py]: Why keeping GroceriesReal(Dataset) class here?

File: pyproject.toml [hunk 3]
```diff
@@ -36,17 +36,13 @@ numpy = ">=1.17 < 1.18"
 nuscenes-devkit = ">=1.0.2 <1.0.3"
```
Comment (adason) [inline pyproject.toml]: Let's remove nuscene dataset here.

File: datasetinsights/datasets/groceries_real.py (hunk 2 above)
Comment (BlairLee) [reply datasetinsights/datasets/groceries_real.py]: Although we don't need this class in general, I think it's the best place to keep it, cause it belongs to the dataset category.

File: datasetinsights/datasets/coco.py (hunk 1 above)
Comment (86sanj) [reply datasetinsights/datasets/coco.py]: As discussed earlier, we are keeping only those dataset subclasses which do not have dependencies on tf and torch. 
- coco is not a subclass of Dataset, its just preprocess the downloaded data.

File: datasetinsights/datasets/groceries_real.py (hunk 2 above)
Comment (86sanj) [reply datasetinsights/datasets/groceries_real.py]: It does not have tf and torch dependencies so I have not removed.

File: pyproject.toml (hunk 3 above)
Comment (86sanj) [reply pyproject.toml]: FYI: nuscenes-devkit has dependencies/includes following components:
  - nuscenes and kitty dataset
  - scikit-learn and opencv-python

File: datasetinsights/datasets/coco.py (hunk 1 above)
Comment (adason) [reply datasetinsights/datasets/coco.py]: So, what part of the datasetinsights package do we want users to use? I see inconsistent patterns here for Coco and GroceriesReal dataset. One is still a dataset class the other only have preprocess method and capability to load coco dataset from disk. 

I suggest we think about the design of dataset class (and dataset catalog object) more carafully and decide what module we keep and what we remove.

**Comment by adason** [comment]
What about `nyudepth.py`?

File: pyproject.toml [hunk 4]
```diff
@@ -33,21 +33,18 @@ google-cloud-storage = ">=1.24.1 <=1.28.1"
...
 yacs = "^0.1.6"
 pycocotools = "^2.0.0"
 codetiming = "^1.2.0"
 pandas = "^1.0.1"
 tqdm = "^4.45.0"
 dask = {extras = ["complete"], version = "^2.14.0"}
 dash = "1.12.0"
-tensorflow = "^2.2.0"
 click = "^7.1.2"
+scikit-learn = "^0.23.2"
```
Comment (adason) [inline pyproject.toml]: Why are these two requirements added?

File: datasetinsights/datasets/__init__.py [hunk 5]
```diff
@@ -1,16 +1,9 @@
 from .base import Dataset
-from .cityscapes import Cityscapes
 from .coco import CocoDetection
```
Comment (adason) [inline datasetinsights/datasets/__init__.py]: Should we keep CocoDetection?

File: datasetinsights/datasets/__init__.py [hunk 6]
```diff
@@ -1,16 +1,9 @@
...
 from .synthetic import SynDetection2D
 
 __all__ = [
-    "Cityscapes",
     "CocoDetection",
     "Dataset",
-    "GroceriesReal",
-    "GoogleGroceriesReal",
-    "NyuDepth",
     "SynDetection2D",
```
Comment (adason) [inline datasetinsights/datasets/__init__.py]: Should we keep `SynDetection2D` here?

File: pyproject.toml [hunk 7]
```diff
@@ -33,21 +33,18 @@ google-cloud-storage = ">=1.24.1 <=1.28.1"
 jupyter = "^1.0.0"
 kornia = "^0.1.4"
```
Comment (adason) [inline pyproject.toml]: Is this package still required?

File: pyproject.toml [hunk 8]
```diff
@@ -33,21 +33,18 @@ google-cloud-storage = ">=1.24.1 <=1.28.1"
...
 numpy = ">=1.17 < 1.18"
-nuscenes-devkit = ">=1.0.2 <1.0.3"
 plotly = "^4.4.1"
 pyquaternion = "^0.9.5"
-pytorch-ignite = "^0.3.0"
-torch = "~1.4.0"
-torchvision = "~0.5"
 yacs = "^0.1.6"
 pycocotools = "^2.0.0"
```
Comment (adason) [inline pyproject.toml]: Is this still required?

File: pyproject.toml [hunk 9]
```diff
@@ -33,21 +33,18 @@ google-cloud-storage = ">=1.24.1 <=1.28.1"
...
 tqdm = "^4.45.0"
 dask = {extras = ["complete"], version = "^2.14.0"}
 dash = "1.12.0"
-tensorflow = "^2.2.0"
 click = "^7.1.2"
+scikit-learn = "^0.23.2"
+opencv-python = "^4.4.0.42"
 
 
 [tool.poetry.dev-dependencies]
```
Comment (adason) [inline pyproject.toml:48]: Please remove kfp in dev-dependencies

File: docs/requirements.txt [hunk 10]
```diff
@@ -21,110 +20,103 @@ cloudpickle==1.6.0
...
 pycparser==2.20
 pyflakes==2.2.0
-pygments==2.6.1
+pygments==2.7.0
 pyparsing==2.4.7
 pyquaternion==0.9.5
-pyrsistent==0.17.0
-pytest==5.4.3
+pyrsistent==0.17.3
+pytest==6.0.2
```
Comment (BlairLee) [inline docs/requirements.txt:109]: Is there any reason why we upgrade those packages?

**Comment by BlairLee** [comment]
Please remove `.gitlab-ci.yml` cause we no longer need it.

File: pyproject.toml (hunk 4 above)
Comment (86sanj) [reply pyproject.toml]: - Removed scikit-learn: as it was used for train-split but we don't need as we don't model training code.
- We need opencv-python, as it is used in bounding box 2d and 3d and its been used in perception stats notebook.

File: docs/requirements.txt (hunk 10 above)
Comment (86sanj) [reply docs/requirements.txt:109]: Its not not manual update, after removal of poetry dependencies I updated requirement.txt using this: "poetry export --dev --without-hashes -f requirements.txt > docs/requirements.txt"

File: pyproject.toml (hunk 8 above)
Comment (86sanj) [reply pyproject.toml]: Removed pycocotools

File: pyproject.toml (hunk 7 above)
Comment (86sanj) [reply pyproject.toml]: Removed kornia

File: .github/workflows/publish-docker-hub.yaml [hunk 11]
```diff
@@ -15,5 +15,5 @@ jobs:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_ACCESS_TOKEN }}
           repository: unitytechnologies/datasetinsights
-          tags: latest
+          tags: 1.0
```
Comment (Saurav-D) [inline .github/workflows/publish-docker-hub.yaml]: Why do we want to tag this as 1.0 instead of latest?

Comment (86sanj) [reply .github/workflows/publish-docker-hub.yaml]: - To support existing users, latest tag should have model experimentation code until experimentation repo is up and ready. So we don't want latest tag to be created when we release code from master.
- Will create latest tag only when we release from 0.2.x branch.

Comment (adason) [reply .github/workflows/publish-docker-hub.yaml]: I will recommend leave it as latest. Users can still access to docker image with tag 0.2.5, 0.2.6,.... and I see no harm having the `latest` branch always reflect the current master.

File: datasetinsights/datasets/__init__.py [hunk 12]
```diff
@@ -1,16 +1,5 @@
...
-from .cityscapes import Cityscapes
-from .coco import CocoDetection
-from .groceries_real import GoogleGroceriesReal, GroceriesReal
-from .nyudepth import NyuDepth
-from .synthetic import SynDetection2D
 
 __all__ = [
-    "Cityscapes",
-    "CocoDetection",
     "Dataset",
```
Comment (adason) [inline datasetinsights/datasets/__init__.py]: Why is this file still required?

**Review by adason — APPROVED** [review]

**Review by adason — CHANGES_REQUESTED** [review]
can you revert changes to the Perception_Statistics notebook?

**Review by adason — APPROVED** [review]

**Review by Saurav-D — APPROVED** [review]

File: README.md [hunk 13]
```diff
@@ -2,6 +2,7 @@
 
 Unity Dataset Insights is a python package for understanding synthetic datasets.
 This package enables users to analyze synthetic datasets generated using the [Perception SDK](https://github.com/Unity-Technologies/com.unity.perception).
+User can download the data, parse the metadata and analyze on the notebook.
```
Comment (romerocesar) [inline README.md:5]: I think we need to rewrite this whole paragraph. Here's a suggestion in a single sentence:

Unity Dataset Insights is a python package for downloading, parsing and analyzing synthetic datasets generated using the [Unity Perception](https://github.com/Unity-Technologies/com.unity.perception) package.

File: README.md [hunk 14]
```diff
@@ -13,9 +14,9 @@ Dataset Insights maintains a pip package for easy installation. It can work in a
 
 We provide a sample [notebook](notebooks/SynthDet_Statistics.ipynb) to help you get started with dataset statistics for the [SynthDet](https://github.com/Unity-Technologies/SynthDet) project. We plan to support other sample Unity projects in the future.
 
-### Dataset Evaluation
+### Dataset Download
 
-Dataset evaluation provides tools to train and evaluate ML models for different datasets. You can run `download`, `train` and `evaluate` commands:
+Dataset download provides tools to download datasets from HTTP(s), GCS and Unity simulation project . You can run `download` command:
```
Comment (romerocesar) [inline README.md:19]: this makes it sound like "dataset download" is a separate package. Also, it should refer to _the_ command

File: datasetinsights/constants.py [hunk 15]
```diff
@@ -4,17 +4,6 @@
...
-DEFAULT_KFP_UI_METADATA_FILENAME = "mlpipeline-ui-metadata.json"
-
-DEFAULT_TENSORBOARD_LOG_DIR = os.path.join(
-    PROJECT_ROOT, "runs", TIMESTAMP_SUFFIX
-)
-DEFAULT_CHECKPOINT_DIR = os.path.join(
-    PROJECT_ROOT, "checkpoints", TIMESTAMP_SUFFIX
-)
 
 GCS_BUCKET = "thea-dev"
```
Comment (romerocesar) [inline datasetinsights/constants.py:8]: why is this in the public repo?

File: docs/requirements.txt [hunk 16]
```diff
@@ -1,13 +1,12 @@
-absl-py==0.10.0
 alabaster==0.7.12
```
Comment (romerocesar) [inline docs/requirements.txt:1]: why do we need to keep this requirements.txt around if we're using poetry?

Comment (86sanj) [reply docs/requirements.txt:1]: Known issue 2: https://github.com/Unity-Technologies/datasetinsights/tree/master/docs
